haas/api.py

```diff
@@ -510,6 +510,25 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """List all networks.
+    Returns a JSON dictionary of dictionaries, indexed by the name of the network.
+    The interior dictionaries are indexed by project and channel.
+    
+    Example:  {"netA": {"channel": "101", "project": "qproj-01"}}
+    """
+    db = model.Session()
+    networks = db.query(model.Network).all()
```

Comment:
The db should be accessed through local.db (see the above function) to prevent session leakage.

Thus, the `db = ...` line should be deleted and this line should look like `networks = local.db.query(...`